### PR TITLE
Update model/schema of article

### LINKED ISSUE #59: Update model/schema of article
Update following new design

### PR DESCRIPTION
**Pull request by KevinDangVN**
Resolve #59

### CODE REVIEW

File: src/model/asset.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  articleId: Schema.Types.ObjectId;
+  s3link: String;
+  type: String;
+  uri: String;
+  created: Date;
+}
+
+export const SchemaAsset = new Schema({
+  uuid: { type: String, default: uuidv4, unique: true, index: true },
+  articleId: { type: Schema.Types.ObjectId, ref: 'Article' },
```
Comment (chiro-hiro) [inline src/model/asset.ts:30]: Do we need `author` here? I think `author` -> `user` is make sense.

File: src/model/comment.ts [hunk 2]
```diff
@@ -10,8 +10,8 @@ import { v4 as uuidv4 } from 'uuid';
 export interface IDocumentComment extends Document {
   uuid: String;
   author: Schema.Types.ObjectId;
-  article: Schema.Types.ObjectId;
-  reply: Schema.Types.ObjectId;
+  articleId: Schema.Types.ObjectId;
```
Comment (chiro-hiro) [inline src/model/comment.ts:13]: It must be `article` to prevent confusion after we do `populate()`

File: src/model/session.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { v4 as uuidv4 } from 'uuid';
 export interface IDocumentSession extends Document {
   uuid: String;
   user: Schema.Types.ObjectId;
-  device: Schema.Types.ObjectId;
+  deviceId: Schema.Types.ObjectId;
```
Comment (chiro-hiro) [inline src/model/session.ts:12]: Should be `device` please check above comment.

File: src/model/user.ts [hunk 4]
```diff
@@ -8,21 +8,22 @@ import { v4 as uuidv4 } from 'uuid';
  * @extends {Document}
  */
 export interface IDocumentUser extends Document {
-  profile: Schema.Types.ObjectId;
-  device: Schema.Types.ObjectId[];
   uuid: String;
+  profileId: Schema.Types.ObjectId;
```
Comment (chiro-hiro) [inline src/model/user.ts:12]: Please, I've changed it on a purpose

File: src/mux/article.ts [hunk 5]
```diff
@@ -29,21 +29,11 @@ Mux.post<IArticle>(
     const result = await imArticle.save();
     const savedArticle = await ModelArticle.findById(result._id).populate('author', '-_id -profile');
     if (savedArticle) {
-      const { uuid, tags, author, vote, comments, title, content, created, updated } = <IArticle>(
-        savedArticle.toObject()
-      );
+      const responseArticle = <IArticle>savedArticle.toObject();
```
Comment (chiro-hiro) [inline src/mux/article.ts:32]: Don't do this, I've just wanted to return which I prefer to return.

**Comment by chiro-hiro** [comment]
@KevinDangVN I think, you get me wrong. I will close this pull request, the only valuable is asset model that's all we need for now. All other changes were reversed my changes.

File: src/model/asset.ts (hunk 1 above)
Comment (KevinDangVN) [reply src/model/asset.ts:30]: okay, I will change author field to user.

**Pull request closed without merging**